### PR TITLE
Support Node 10

### PR DESCRIPTION
**Pull request by samrg472**
Node 10 dropped support for some deprecated APIs. This updates the API usage to be Node 10 compatible.

This was built and tested on MacOS without issues. It built successfully for all target versions from Node 4.4.3 to Node 10.

For testing: I dropped in the built module into a local project on my system (with Node 10) that uses web sockets and everything worked without an issue (opening/closing/sending text and binary/receiving text and binary).

### CODE REVIEW

**Comment by alexhultman** [comment]
Sweet work, will merge and check out before next version

File: nodejs/Makefile [hunk 1]
```diff
@@ -25,10 +26,11 @@ targets:
 	curl https://nodejs.org/dist/v7.1.0/node-v7.1.0-headers.tar.gz | tar xz -C targets
 	curl https://nodejs.org/dist/v8.1.2/node-v8.1.2-headers.tar.gz | tar xz -C targets
 	curl https://nodejs.org/dist/v9.2.0/node-v9.2.0-headers.tar.gz | tar xz -C targets
+	curl https://nodejs.org/dist/v10.0.0/node-v10.0.0-headers.tar.gz | tar xz -C targets
 Linux:
 	g++ $(CPP_SHARED) -static-libstdc++ -static-libgcc -I $$NODE/include/node -s -o dist/uws_linux_$$ABI.node
 Darwin:
-	g++ $(CPP_SHARED) $(CPP_OSX) -I $$NODE/include/node -o dist/uws_darwin_$$ABI.node
+	g++ $(CPP_SHARED) $(CPP_OSX) -I /usr/local/opt/openssl/include -I $$NODE/include/node -o dist/uws_darwin_$$ABI.node
```
Comment (alexhultman) [inline nodejs/Makefile]: This is not correct, need to link to the openssl that comes with each node version

Comment (samrg472) [reply nodejs/Makefile]: Sorry about that, I forgot to remove it when I committed.

**Comment by alexhultman** [comment]
You can "fix" all those warnings by adding {0, 0} as last argument to node::MakeCallback but I benchmarked this (and variations) and it lowers perf. significantly.

The Node.js geniuses have achieved the impossible: make MakeCallback _EVEN_ slower and _EVEN_ more bloated... holy shit it gets worse every release!

**Comment by alexhultman** [comment]
Node.js "core members" are complete idiots, complete and utter morons

**Comment by fanatid** [comment]
@alexhultman when you planning publish this changes?

**Comment by alexhultman** [comment]
probably 14 days give or take

**Comment by gjavi1** [comment]
@alexhultman is the npm versioning wrong at the moment? running `npm view uws versions` gives you this:
```
[ '0.14.1',
  '0.14.3',
  '0.14.4',
  '0.14.5',
  '8.14.0',
  '8.14.1',
  '9.14.0',
  '9.147.0',
  '9.148.0' ]
```
Is the jump from `9.14.0` to `9.147.0` a typo issue.

**Comment by bradennapier** [comment]
Any way this can expedite?  Why 14 days?  Since this doesnt appear to be simple in any way to do with how the repo is setup - woudl be nice to get v10 support as this is a huge blocker for us

**Pull request merged**